src/base/17.0.1.3/attr_domains2expr.py

```diff
@@ -319,9 +319,12 @@ def fix_elem(cr, model, elem, comb_arch):
                 extra_invalid,
             )
         for key in extra:
-            value = ast.unparse(attrs[key])
-            _logger.info("Inlined %s=%r", key, value)
-            elem.set(key, value)
+            try:
+                value = ast.unparse(attrs[key])
+                elem.set(key, value)
+                _logger.info("Inlined %s=%r", key, value)
+            except ValueError:
+                _logger.exception("Skipping invalid attribute name %s in `attrs`", key)
```

Comment:
```suggestion
            value = ast.unparse(attrs[key])                
            try:
                etree.QName(key)
            except ValueError:
                _logger.error("Skipping invalid attribute name %r in `attrs` with value: %s", key, value)
            else:
                elem.set(key, value)
                _logger.info("Inlined %s=%r", key, value)
```

Reply:
This works as well, and is probably safer since `QName()` calls the same validator function (`_pyXmlNameIsValid()`) as `set()` but only looks at the key. 
Done. :+1: